File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Prompt for change owed
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change <= 0);

    int cents = round(change * 100);
    int coins = 0;

    // quarters
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }

    // dimes
    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }

    //nickels
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }

    // pennies
    while (cents >= 1)
    {
        cents -= 1;
        coins++;
    }


    // Print amount of coins
    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float, get_int, get_string

# Prompt for change owed


def get_positive_float():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
    return n


change = get_positive_float()
cents = round(change * 100)
coins = 0

# quarters
while cents >= 25:
    cents -= 25
    coins += 1

# dimes
while cents >= 10:
    cents -= 10
    coins += 1

# nickels
while cents >= 5:
    cents -= 5
    coins += 1

# pennies
while cents >= 1:
    cents -= 1
    coins += 1


# Print amount of coins
print(coins)